cmd/juju/application/utils/origin.go

```diff
@@ -36,22 +35,49 @@ func DeduceOrigin(url *charm.URL, channel charm.Channel, platform corecharm.Plat
 		architecture = arch.DefaultArchitecture
 	}
 
-	var origin commoncharm.Origin
+	var (
+		origin commoncharm.Origin
+		series string
+		err    error
+	)
+	if platform.OS != "" && platform.Channel != "" {
+		if platform.OS == "kubernetes" || platform.Channel == "kubernetes" {
+			series = "kubernetes"
+		} else {
+			series, err = coreseries.GetSeriesFromBase(coreseries.Base{Name: platform.OS, Channel: platform.Channel})
+			if err != nil {
+				return commoncharm.Origin{}, errors.Trace(err)
+			}
+		}
+	} else if platform.OS == "" && platform.Channel != "" {
+		// If there is a series, ensure there is an OS.
+		series, err = coreseries.VersionSeries(platform.Channel)
```

Comment:
In what case would we have a channel but no OS?

Reply:
We wouldn't, hence the checks are somewhat obsolete. They were written for dealing with series etc